packages/engine-core/src/data-proxy/utils/request.ts

```diff
@@ -70,6 +73,7 @@ function buildOptions(options: RequestOptions): Https.RequestOptions {
  */
 function buildResponse(incomingData: Buffer[], response: IncomingMessage): RequestResponse {
   return {
+    text: () => Buffer.concat(incomingData).toString(),
     json: () => JSON.parse(Buffer.concat(incomingData).toString()),
```

Comment:
Should we extract the common expression into a variable? Probably doesn't matter much in practice, but still feels wrong to perform these operations twice.

Reply:
Not sure, it's very useful here. Here we need to have a function as it's only done "on demand" and it's very minimal.

Reply:
Ah, sorry, I didn't notice these were functions.

Reply:
I was reading too fast I guess 😅